contractcourt/channel_arbitrator.go

```diff
@@ -244,11 +260,44 @@ func (c *ChannelArbitrator) Start() error {
 		return err
 	}
 
+	// Check the database for the close status of this channel.
+	isClosing, closeType, err := c.cfg.IsPendingClose()
+	if err != nil {
+		c.cfg.BlockEpochs.Cancel()
+		return err
+	}
+
+	// If the channel has been marked pending close in the database, and we
+	// haven't transitioned the state machine to StateContractClosed (or a
+	// suceeding state), then a state transition most likely failed. We'll
+	// try to recover from this by manually advancing the state by setting
+	// the corresponding close trigger.
+	trigger := chainTrigger
+	if isClosing {
+		switch c.state {
+		case StateDefault:
+			fallthrough
+		case StateBroadcastCommit:
+			fallthrough
+		case StateCommitmentBroadcasted:
+			switch closeType {
+			case channeldb.LocalForceClose:
+				trigger = localCloseTrigger
+			case channeldb.RemoteForceClose:
+				trigger = remoteCloseTrigger
+			}
+
+			log.Warnf("ChannelArbitrator(%v): detected stalled "+
```

Comment:
i think this would be clearer as:
```golang
var trigger = chainTrigger
if isClosing {
    if c.state == StateCommitmenBroadcasted {
        switch closeType {
        case channeldb.LocalForceClose:
            trigger = localCloseTrigger
	case channeldb.RemoteForceClose:
	    trigger = remoteCloseTrigger
	}
    }
    log.Warnf()
}
```
the fallthroughs are harder to follow imo

Reply:
It's not equivalent, were you thinking
```go
var trigger = chainTrigger
if isClosing {
    if c.state == StateDefault
         || c.state == StateBroadcastCommit
         || c.state == StateCommitmenBroadcasted {
        switch closeType {
        case channeldb.LocalForceClose:
            trigger = localCloseTrigger
	case channeldb.RemoteForceClose:
	    trigger = remoteCloseTrigger
	}
    }
    log.Warnf()
}
```

Reply:
ah right, forgot that local/remote could be delivered in first two, nvm :)